webapp-django/crashstats/crashstats/models.py

```diff
@@ -1239,11 +1256,17 @@ class RawCrash(SocorroMiddleware):
     )
 
     def get(self, **kwargs):
-        format = kwargs.get('format', 'meta')
-        if format == 'raw_crash':
-            format = kwargs['format'] = 'raw'
-        kwargs['expect_json'] = format != 'raw'
-        return super(RawCrash, self).get(**kwargs)
+        fmt = kwargs.get('format', 'meta')
+        if fmt == 'raw_crash':
+            # legacy
+            fmt = kwargs['format'] = 'raw'
+        expect_dict = fmt != 'raw'
+        result = super(RawCrash, self).get(**kwargs)
+        # This 'result', will either be a binary blob or a python dict.
+        # Unless kwargs['format']==raw, this has to be a python dict.
+        if expect_dict and not isinstance(result, dict):
+            raise BadArgumentError(fmt)
```

Comment:
This will show an error saying that the value of ``fmt`` is a field that has a bad value. Instead I think this should say ``raise BadArgumentError('format')``.